src/main/java/com/netchill/services/streaming/TrailerService.java

```diff
@@ -0,0 +1,44 @@
+package com.netchill.services.streaming;
+
+import com.netchill.api.moviedb.models.Trailer;
+import com.netchill.api.moviedb.models.TrailerKey;
+import com.netchill.api.moviedb.services.trailer.TrailerApiService;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import java.util.List;
+
+@Singleton
+public class TrailerService {
+    private TrailerApiService trailerApiClient;
+    private static final String TRAILER_TYPE = "Trailer";
+
+    @Inject
+    private TrailerService(TrailerApiService trailerApiClient) {
+        this.trailerApiClient = trailerApiClient;
+    }
+
+    public TrailerKey getSerieTrailerById(Long id) {
+        Trailer trailer = trailerApiClient.getSerieTrailerById(id);
+        return this.getTrailer(trailer.getKeyList());
+    }
+
+    public TrailerKey getMovieTrailerById(Long id) {
+        Trailer trailer = trailerApiClient.getMovieTrailerById(id);
+        return this.getTrailer(trailer.getKeyList());
+    }
+
+    private TrailerKey getTrailer(List<TrailerKey> trailerKeys) {
+        if (trailerKeys.isEmpty()) {
+            return new TrailerKey();
+        }
```

Comment:
C'est un peu bizarre, si la Serie n'a pas de trailer pourquoi tu crée un TrailerKey vide ?

Si c'est un cas envisageable, donc pas une erreur, ta méthode `getTrailer` (qu'il faut renommer, soit dite en passant) devrait retourner un Optional.


Exemple :

```suggestion
        if (trailerKeys.isEmpty()) {
            return Optional.empty();
        }
```

Reply:
tu ne m'a pas répondu

Reply:
Je renvoyais un TrailerKey vide pour que dans le front je dis que si la key est null alors j'affiche une phrase. La key étant un attribut de l'objet TrailerKey.

Reply:
Ah ok, mais ce n'est pas une très bonne conception, ton back dépend de ton front alors que ça doit être l'inverse.

Reply:
tu devrais envoyer une erreur (exception) et en front si tu reçois une erreur alors tu affiches ton message

Reply:
Ah oui ok !